tests/ui/default_box_assignments.rs

```diff
@@ -0,0 +1,17 @@
+#![warn(clippy::default_box_assignments)]
+
+fn main() {
+    let mut b = Box::new(1u32);
+    b = Default::default();
+    //~^ default_box_assignments
+    b = Box::default();
+    //~^ default_box_assignments
+
+    // No lint for assigning to the storage
+    *b = Default::default();
+    *b = u32::default();
+
+    // No lint for assigning to Box<T> where T is unsized
```

Comment:
Is it really because `T` is unsized, or because `T` does not implement `Default` why `Box<T>` does?

Reply:
It’s because `T` doesn’t implement `Default`. My initial implementation checked whether `T` implemented `Sized`, but I figured that I should be checking for a `Default` impl. I’ve changed this comment now.